### PR TITLE
Use a separate class for M2L translation and change expansion default to FFT

### PR DESCRIPTION
**Pull request by isuruf**
Fixes https://github.com/inducer/sumpy/issues/104

- [x] Needs #117 
- [x] Needs #115
- [ ] Mutual dependency on inducer/pytential#151

### CODE REVIEW

File: test/test_fmm.py [hunk 1]
```diff
@@ -193,11 +193,17 @@ def test_sumpy_fmm(ctx_factory, knl, local_expn_class, mpole_expn_class,
         else:
             translation_classes_data = None
 
+        if use_fft:
+            from sumpy.expansion.m2l import VolumeTaylorM2LWithFFT
+            m2l_translation = VolumeTaylorM2LWithFFT()
+        else:
+            m2l_translation = None
```
[the remark this answers is not in the document]
Comment (isuruf) [reply test/test_fmm.py]: What do you think of 72b4875?

**Comment by inducer** [comment]
Could you take a look? This now seems to fail CI.

**Comment by isuruf** [comment]
This is because I changed the default for the M2L translation to FFT and now all tests require the additional translation_classes data. How about I keep the default to non FFT in sumpy and use the FFT in pytential where the translation_classes is computed and passed?

**Comment by inducer** [comment]
I'd prefer to have the default at FFT while modifying all the constructor calls in sumpy. I could also live with a non-FFT default that yells about its deprecation.

**Comment by isuruf** [comment]
Factor of 2 change fixed the tests. Once inducer/pytential#102 is merged this should be ready for review.

**Comment by inducer** [comment]
> Factor of 2 change fixed the tests.

Out of curiosity: where was that factor of 2?

Also, for my peace of mind: There are no functional changes here, correct?

**Comment by isuruf** [comment]
> Out of curiosity: where was that factor of 2?

https://github.com/inducer/sumpy/pull/113/files#diff-1ba22e3d79b16504929f9dbd8675d879c2b6e19b498dbddbd29cdc49bec1797eR327

> Also, for my peace of mind: There are no functional changes here, correct?

There's the factor of 2 change in rscale and the change of default for M2L translation type to FFT.

**Review by inducer — APPROVED** [review]
This was hard to review because of the amount of code that's moved. But pending your confirmation that there are no functional changes and that the factor of 2 is innocuous, this LGTM. Thanks for working on this!

**Comment by inducer** [comment]
Didn't see your response until after I submitted the review. Thanks for the summary! LGTM, in it goes.

**Comment by inducer** [comment]
(And thanks for working on this!)

**Pull request merged**